File: typegate/tests/e2e/typegraph/typegraphs/python/multiple_runtimes.py
from typegraph_next import Graph, t, typegraph
from typegraph_next.policy import Policy
from typegraph_next.runtimes.deno import DenoRuntime
from typegraph_next.runtimes.python import PythonRuntime


@typegraph()
def test_multiple_runtimes(g: Graph):
    public = Policy.public()
    deno = DenoRuntime()
    python = PythonRuntime()

    g.expose(
        add=python.from_lambda(
            t.struct(
                {
                    "first": t.float(),
                    "second": t.float(),
                }
            ),
            t.float(),
            lambda x: x["first"] + x["second"],
        ).with_policy(public),
        multiply=deno.func(
            t.struct({"first": t.float(), "second": t.float()}),
            t.float(),
            code="({ first, second }) => first * second",
        ).with_policy(public),
    )

File: tests/e2e/typegraph/typegraphs/deno/multiple_runtimes.ts
// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
// SPDX-License-Identifier: MPL-2.0

import { Policy, t, typegraph } from "@typegraph/sdk/index.ts";
import { DenoRuntime } from "@typegraph/sdk/runtimes/deno.ts";
import { PythonRuntime } from "@typegraph/sdk/runtimes/python.ts";

typegraph("test-multiple-runtimes", (g: any) => {
  const pub = Policy.public();
  const deno = new DenoRuntime();
  const python = new PythonRuntime();

  g.expose({
    add: python
      .fromLambda(
        t.struct({ first: t.float(), second: t.float() }),
        t.float(),
        { code: "lambda x: x['first'] + x['second']" },
      )
      .withPolicy(pub),
    multiply: deno
      .func(t.struct({ first: t.float(), second: t.float() }), t.float(), {
        code: "({first, second}) => first * second",
      })
      .withPolicy(pub),
  });
});
